### PR TITLE
build: migrate to Nx v17 inc deps

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR updates Nx packages. In addition to the new package versions a couple of refactoring's needed to be done to make it work.

**Included changes:**
- package updates
- refactoring of the `nx-plugin` code due to changes of the `nx.json` structure
- refactorings of the `plugin-eslint` package due to dependency changes

**Included changes to discuss:**
- the current PR includes a bigger refactoring of the tests in `package-eslint` 
- the number of rules got reduces
- test setup per case

### CODE REVIEW

**Comment by BioPhoton** [comment]
I don't really get why the eslint-plugin tests fail... @vmasek any clue?

**Comment by BioPhoton** [comment]
I know now why the tests where failing. They rely on real eslint setup and the ouatput data changed.
@matejchalk I already created #255 and #254 and updated #224 

What would be the minimum fix to make this PR turn 🟢?

**Comment by BioPhoton** [comment]
I wanted to get the update in before, but also decided to day that its not a good idea.

**Comment by BioPhoton** [comment]
Ok, I gave the test refactoring a try based on our discussions with @Tlacenka. It's not perfect but much more entangles and smaller in rules. 

@matejchalk let me know if I can polish this version or if I can adopt it.

**Review by matejchalk — CHANGES_REQUESTED** [review]
Sorry, but the PR scope is too big. Refactoring the `plugin-eslint` tests isn't necessary to update Nx. Please revert those changes and just run `npx nx test plugin-eslint -u` to update the snapshot.

The fact that the tests failed after a major update isn't a bad thing, in my opinion. It gives us visibility into the impact of those changes.

File: nx.json [hunk 1]
```diff
@@ -28,29 +15,37 @@
...
         "maxWarnings": 0,
         "reportUnusedDisableDirectives": "warn"
-      }
+      },
+      "cache": true
+    },
+    "@nx/vite:test": {
+      "cache": true,
+      "inputs": ["default", "^production"]
     },
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply nx.json:24]: that was done by the Nx migration. They changed the targets and caching definition. Both changes are in `nx.json`.

I had no time to merge that with @Tlacenka's changes so far

**Comment by BioPhoton** [comment]
@matejchalk 
I will first refactor the plugin-eslint tests and then update.

File: migrations.json
[the remark this answers is not in the document]
Comment (BioPhoton) [reply migrations.json:1]: Yes! Also documented here:
https://nx.dev/core-features/automate-updating-dependencies#step-3-cleaning-up

[the remark this answers is not in the document]
Comment (BioPhoton) [reply migrations.json:1]: Just learned that it is not needed anymore

**Comment by matejchalk** [comment]
> I will first refactor the plugin-eslint tests and then update Nx.

But why? :sweat_smile: The priorities are completely the opposite in my opinion. I'm not even convinced the ESLint tests need to be refactored (would be good to get @Tlacenka's opinion on this), and it surely isn't a blocker for updating Nx (`npx nx test plugin-eslint -u` is the only necessary step, I believe).

I honestly don't like having to decline your PRs, but when you decide to make big chances before first consulting it with us (e.g. on refinement), then I'm afraid it's gonna keep happening :disappointed:

**Comment by Tlacenka** [comment]
> I'm not even convinced the ESLint tests need to be refactored (would be good to get @Tlacenka's opinion on this)

The `plugin-eslint` package in the original version is in my opinion sufficiently covered  by both unit (internal logic) and integration (initiation and executing the runner) tests.

I am unfortunately unable to determine which changes are truly necessary due to the updated `nx` version. The ESLint rules and results seem to be different, variables renamed.

Should the changes be done as a part of #255 , they should in my opinion be first investigated in detail and any targetted changes should be scoped and prioritised separately (otherwise the issue could get bloated easily), not as a part of this PR.

**Pull request closed without merging**